### PR TITLE
APPS-1015 Refactor REST API of the service

### PR DESCRIPTION
**Pull request by korotkov-aerospike**

### CODE REVIEW

File: internal/server/restore_handlers.go [hunk 1]
```diff
@@ -33,9 +33,13 @@ func (ws *HTTPServer) restoreFullHandler(w http.ResponseWriter, r *http.Request)
 			http.Error(w, err.Error(), http.StatusBadRequest)
 			return
 		}
-		restoreRequest := request.ToRestoreRequest()
-		jobID := ws.restoreService.Restore(&restoreRequest)
-		slog.Info("Restore action", "jobID", jobID, "request", request)
+		requestInternal := &model.RestoreRequestInternal{
```
Comment (reugn) [inline internal/server/restore_handlers.go:36]: Can be done via `ToInternal` method of `RestoreRequest`

File: pkg/model/restore_request.go [hunk 2]
```diff
@@ -2,44 +2,30 @@ package model
...
+	SourceStorage     *Storage          `json:"source,omitempty"`
 }
 
-// RestoreIncrementalRequest represents an incremental restore operation request.
-type RestoreIncrementalRequest struct {
-	DestinationCuster *AerospikeCluster `json:"destination,omitempty"`
-	SourceStorage     *Storage          `json:"source,omitempty"`
-	Policy            *RestorePolicy    `json:"policy,omitempty"`
-	File              *string           `json:"file,omitempty"`
+// RestoreRequestInternal is only used internally to pass data to restore library.
```
Comment (reugn) [inline pkg/model/restore_request.go:14]:
```suggestion
// RestoreRequestInternal is used internally to prepopulate data for the restore operation.
```

File: pkg/service/backup_scheduler.go [hunk 3]
```diff
@@ -133,30 +144,32 @@ func (h *BackupHandler) scheduleBackupPeriodically(
...
-		slog.Debug("Release fullBackupInProgress lock", "name", h.backupRoutine.Name)
+		slog.Debug("Release fullBackupInProgress lock", "name", h.routineName)
 	}()
 
 	backupRunFunc := func() {
 		started := time.Now()
+		slog.Debug("Starting full backup", "name", h.routineName)
 		stats := backupService.BackupRun(h.backupRoutine, h.backupPolicy, h.cluster, h.storage, shared.BackupOptions{})
 		if stats == nil {
+			slog.Warn("Failed full backup", "policy", h.routineName)
```
Comment (reugn) [inline pkg/service/backup_scheduler.go:172]:
```suggestion
			slog.Warn("Failed full backup", "name", h.routineName)
```

File: pkg/service/backup_scheduler.go [hunk 4]
```diff
@@ -179,31 +192,33 @@ func (h *BackupHandler) runFullBackup(now time.Time) {
...
 	}
 	backupRunFunc := func() {
 		opts := shared.BackupOptions{}
 		lastIncrRunEpoch := state.LastIncrRun.UnixNano()
 		opts.ModAfter = &lastIncrRunEpoch
 		started := time.Now()
+		slog.Debug("Starting incremental backup", "name", h.routineName)
 		stats := backupService.BackupRun(h.backupRoutine, h.backupPolicy, h.cluster, h.storage, opts)
 		if stats == nil {
+			slog.Warn("Failed incremental backup", "policy", h.routineName)
```
Comment (reugn) [inline pkg/service/backup_scheduler.go:221]:
```suggestion
			slog.Warn("Failed incremental backup", "name", h.routineName)
```

File: pkg/service/configuration_service_policy_test.go [hunk 5]
```diff
@@ -51,34 +50,29 @@ func TestPolicy_AddErrors(t *testing.T) {
...
 
-	updatedPolicy.Name = &name
-	err = UpdatePolicy(config, updatedPolicy)
+	err = UpdatePolicy(config, &name, &model.BackupPolicy{})
 	if err != nil {
 		t.Errorf("UpdatePolicy failed, expected nil error, got %v", err)
 	}
 
-	if *config.BackupPolicies[name].Name != *updatedPolicy.Name {
+	if *config.BackupPolicies[name] != *(&model.BackupPolicy{}) {
```
Comment (reugn) [inline pkg/service/configuration_service_policy_test.go:66]:
```suggestion
	if *config.BackupPolicies[name] != (model.BackupPolicy{}) {
```

File: pkg/service/configuration_service_routine.go [hunk 6]
```diff
@@ -1,21 +1,14 @@
...
 // AddRoutine
 // adds a new BackupRoutine to the configuration if a routine with the same name doesn't already exist.
-func AddRoutine(config *model.Config, newRoutine *model.BackupRoutine) error {
-	if newRoutine.Storage == "" {
-		return errors.New("storage is empty")
-	}
-	if newRoutine.SourceCluster == "" {
-		return errors.New("cluster is empty")
-	}
+func AddRoutine(config *model.Config, name *string, newRoutine *model.BackupRoutine) error {
```
Comment (reugn) [inline pkg/service/configuration_service_routine.go:11]: You always dereferene the name argument, so it's better to take it by copy as a parameter (in all places).

**Comment by korotkov-aerospike** [comment]
http://3.65.168.247:8080/ui/#pitr/launches/all/3868
all tests passed (I updated test project)

**Pull request merged**